### ISSUE TITLE
Incorrect number of references displayed on Godot project

### ISSUE THREAD

**Issue by drcxd** [razzmatazz/csharp-language-server#137]
![nor](https://github.com/razzmatazz/csharp-language-server/assets/24535087/760b3abe-4075-4fd1-afc7-bdf56e26aa5d)

As the picture shows, the number of refernces displayed is incorrect. This is the only file in the project, so everything gets referenced inside this file. I can guarantee there is no other references. In fact, if I try to find references for a certain variable/function, it gives the correct number of references. Adding new references or removing existing references is reflected correctly on the number of references though the total number of references is not correct.

The content of the `.csproj` file:

```
<Project Sdk="Godot.NET.Sdk/4.2.1">
  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <TargetFramework Condition=" '$(GodotTargetPlatform)' == 'android' ">net7.0</TargetFramework>
    <TargetFramework Condition=" '$(GodotTargetPlatform)' == 'ios' ">net8.0</TargetFramework>
    <EnableDynamicLoading>true</EnableDynamicLoading>
  </PropertyGroup>
</Project>
```

The source code file:

```
using Godot;
// using System;

public partial class player : Area2D
{
    [Export]
    public int Speed { get; set; } = 400;

    public Vector2 ScreenSize;

    // Called when the node enters the scene tree for the first time.
    public override void _Ready()
    {
        ScreenSize = GetViewportRect().Size;
    }

    // // Called every frame. 'delta' is the elapsed time since the previous frame.
    // public override void _Process(double delta)
    // {
    // }
}
```

Note that this issue only occurs in a Godot project, I have tested a clean project that does not depends on Godot, and there is no such issue.

**Comment by Dit05** [comment]
Godot does some code generation to your Node-derived scripts when using Godot Mono. (that's why you need to mark your types `partial`)

Perhaps the extra references are coming from Godot Mono generated code? Try creating a blank project and copying your script over into it. Is the reference count correct **before ever** building the new project for the first time?

**Comment by razzmatazz** [comment]
I *believe* this has been fixed by @tcx4c70 in this commit: https://github.com/razzmatazz/csharp-language-server/commit/d9c42bbb49cfb1aac720d0801b191558debe0f43 (to be released with 0.14.0)
